proposals/00x-azure-workload-identity/README.md

```diff
@@ -0,0 +1,236 @@
+```
+
+> Note: the above example is just for illustration. The plugin you installed is likely to be different.
+
+## Alternatives considered
+
+The first question to ask is whether the `spin azure` plugin is necessary and can it be part of the spin operator. An early plan was to include the Azure pod labels and sericeAccountName in the SpinApp resource specification. When the SpinApp is created, the operator will create the Azure resources and configure the pod with the Azure Workload Identity annotations.
+
+The reason why we don't want to do that is two folds:
+1. The spin operator should remain provider agnostic. It should not be concerned with the underlying cloud provider - Azure or otherwise.
+2. More importantly, there are security concerns with the operator being able to create Azure Managed Identities and Federated Credentials.
+
+Thus, we need to keep the changes to the operator minimal and move the Azure-specific changes to a Spin plugins.
+
+The `spin kube` plugin might be a good candidate, but as the name suggests, it is a plugin for interacting with Kubernetes resources. It is not a good candidate for interacting with Azure resources. It is clear to me that a Azure-specific plugin is necessary to manage Azure resources, such as Authenticating with Azure, creating AKS clusters, creating managed identities, and assigning roles.
+
+The `spin kube scaffold` command, however, is still useful for generating the SpinApp that has the necessary Azure labels and serviceAccountName. I have considered the alternative of having `spin kube` remain provider agnostic and have `spin azure` to be responsible for generating a SpinApp that has the necessary Azure labels. I didn't go with that because I didn't have time to implement it. But if folks think that's a good idea, I can give it a try.
+
+Below is the original proposal for the `workloadIdentity` field in the SpinApp resource specification, which I rejected due to the security concerns mentioned above:
+
+The `workloadIdentity` field has the following structure:
+
+- `workloadIdentity`: Configuration for Workload Identity
+  - `serviceAccountName`: The name of the Kubernetes service account to use for workload identity.
+  - `providerMetadata`: A map of cloud provider-specific configuration.
+
+The Azure provider metadata has the following structure:
+
+- `azure`: true | false
+
+When this field is configured, the operator will:
+1. Mutate the deployment to use the workload identity service account
+2. Label the pod with Azure Workload Identity annotations
+
+Here's an example of a SpinApp using Azure Workload Identity (some fields are omitted for brevity):
+```yaml
+apiVersion: core.spinkube.dev/v1alpha1
+kind: SpinApp
+metadata:
+  name: wid-spinapp
+spec:
+  image: "ghcr.io/spinkube/containerd-shim-spin/examples/spin-rust-hello:v0.13.0"
+  executor: containerd-shim-spin
+  workloadIdentity:
+    serviceAccountName: "workload-identity-sa"
+    providerMetadata:
+      azure: true
```

Comment:
I think this seems reasonable - do we have a sense of what alternative cloud workload identity would look like here? (I haven't spent a huge amount of time on cloud k8s since these became common, so don't have good intuition about what EKS/GKE would look like)